node/subsystem/dispatch-gen/src/lib.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use proc_macro2::TokenStream;
+use quote::{quote, ToTokens};
+use std::fmt;
+use syn::{parse2, Error, Fields, FieldsNamed, FieldsUnnamed, Ident, ItemEnum, Path, Result, Type, Variant};
+
+#[proc_macro_attribute]
+pub fn subsystem_dispatch_gen(attr: proc_macro::TokenStream, item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+	let attr = proc_macro2::TokenStream::from(attr);
+	let item = proc_macro2::TokenStream::from(item);
+	proc_macro::TokenStream::from(subsystem_dispatch_gen2(attr, item))
```

Comment:
Can we please drop this random `2` postfix?

And you can just put the content of `subsystem_dispatch_gen2` into this function.

And you can also just call `into` instead of writing out the full type.

Reply:
If I just make it all spaghetti, then I have to be more careful on the return type, so the indirection keeps the error handling cleanly out of there.